### PR TITLE
😋feat: add type hint for createAction

### PR DESCRIPTION
**Pull request by winrey**

### CODE REVIEW

File: src/createActions.ts [hunk 1]
```diff
@@ -1,23 +1,24 @@
-import { IActions, IActionTypes } from './typings';
+import { IActions, IActionTypes, IActionsTypes } from './typings';
 import createAction from './createAction';
 
 /**
  * generate multiple Flux Standard Action creators
  * @param {IActionTypes} actionTypes
  * @returns {IActions<any>}
```
Comment (taoweicn) [inline src/createActions.ts]: Please modify the function signature at the same time.

File: src/createActions.ts [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+      throw new Error('Action types must be an object!');
+    }
+    const types = Object.keys(actionTypes);
+    if (types.length === 0) {
+      throw new Error('Action types must be not empty!');
+    }
+    return types.reduce((acc: IActions<any>, type: string) => {
+      acc[type.toLowerCase().replace(/([-_]\w)/g, g => g[1].toUpperCase())] = createAction(actionTypes[type]);
+      return acc;
+    }, {}) as unknown as ({} extends T ? IActions<any> : IActionsTypes<T>);
```
Comment (taoweicn) [inline src/createActions.ts]: What's the role of `as unknown`?

File: test/createActions.test.ts [hunk 3]
```diff
@@ -31,6 +31,25 @@ describe('test the generating actions function', () => {
...
+    const payload = { data: 2 };
+    const expectedActions = {
+      request: { type: actionTypes.REQUEST, payload },
+      requestSuccess: { type: actionTypes.REQUEST_SUCCESS, payload },
+      requestFailure: { type: actionTypes['REQUEST-FAILURE'], payload },
+    };
+
+    expect(generatedActions.request(payload)).toEqual(expectedActions.request);
+    expect(generatedActions.requestSuccess(payload)).toEqual(expectedActions.requestSuccess);
+  });
```
Comment (taoweicn) [inline test/createActions.test.ts]: Should check the `generatedActions`'s `requestFailure` property as well, please help me add it to the above test case too.

File: src/createActions.ts (hunk 1 above)
Comment (winrey) [reply src/createActions.ts]: 😂Ohhhh, It's my mistake. I forgot it at all...

File: test/createActions.test.ts (hunk 3 above)
Comment (winrey) [reply test/createActions.test.ts]: My pleasure.

File: src/createActions.ts (hunk 2 above)
Comment (winrey) [reply src/createActions.ts]: To dealing with the annoying type check.
It may have better way after consideration...

**Pull request merged**